### DISCUSSION TITLE
Why is there a catch-all try-except block in cog_unload?

### DISCUSSION THREAD

**Discussion by WitherredAway** [Rapptz/discord.py#9421] [Q&A]
https://github.com/Rapptz/discord.py/blob/e870bb1335e3f824c83a40df4ea9b17f215fde63/discord/ext/commands/cog.py#LL751C1-L754C8
Why is there a try-except here that eats up all exceptions? Isn't this bad practice? I was trying to do something in cog_unload and wasted like an hour trying to figure out why it's not working just to find that all the exceptions are being ignored :(

**Comment by tailoric** [comment]
All I can find about it is this reasoning in this issue:
https://github.com/Rapptz/discord.py/issues/6113#issuecomment-738450803
